### PR TITLE
Jg motor control

### PR DESCRIPTION
**Pull request by Jaden803**

### CODE REVIEW

File: common/core/motors/dcm_control_test.cc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    FakeInit(&motor, &motor_ctrl);
+    QEnc_Init(&qenc, &qenc_ctrl, &motor);
+    DCM_Control_Init(&dev, &motor_ctrl, &qenc_ctrl);
+    FakeSetEnable(&motor, true);
+    config(&dev);
+    EXPECT_TRUE(qenc_ctrl.command_rotate(&qenc_ctrl, 10));
+    EXPECT_EQ(qenc_ctrl.state, idle);
+    motor_update(&dev);
+    EXPECT_EQ(qenc_ctrl.state, rotating);
+    EXPECT_TRUE(dev.cmd_enable);
```
Comment (nfan17) [inline common/core/motors/dcm_control_test.cc]: okay, we're heading more towards the correct direction here - what we need to test with this unit test is that when we rotate the encoder to read less than the commanded angle (in this case 10), the code should still say it's driving the motor. Then, if we command it to rotate again past that threshold, it should say it has stopped.

we also want to test different edge cases, such as if the encoder overflows during the time that it is rotating. a good test is probably to have a loop that increments the encoder by a small amount each iteration, and then keeps a counter - then we expect the counter to be equal the number of iterations needed to pass the degree commanded. We'd want to do this in both directions, and through overflow

File: common/core/motors/DC/drv8242_dcm.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (nfan17) [inline common/core/motors/DC/drv8242_dcm.h]: thinking more about it, I think we should probably just ditch this piece of code - unfortunately it's a bit of a waste but it means we'd have to maintain it without actual hardware to test on to verify it works, so I'd rather just delete it
